Sources/App/Core/Search.swift

```diff
@@ -95,16 +99,109 @@ enum Search {
             .map { $0.replacingOccurrences(of: "]", with: "\\]") }
             .filter { !$0.isEmpty }
     }
+    
+    struct SearchFilter {
```

Comment:
Generally speaking, I'm still not happy with this type and it's contents.

I'm thinking it might be best as an enum to give us a little more safety in terms of filter input and query outputs (there's quite a few routes where you can provide an invalid value which passes some checks but not both and is confusing elsewhere in the app)

Equally, it might sense to create the SQL expressions as part of the initialiser, and just bind them to the predicate builder at that point. This would prevent a lot of the duplication at least. Leaning towards this solution at the moment but keeping an open mind -- as I'm sure you can see by the commits I've gone through a few different solutions as I realise that different filters require wildly different queries.